docs/contributing.rst

```diff
@@ -312,6 +312,33 @@ Please:
 .. _PEP 8 - Style Guide for Python Code:
   https://www.python.org/dev/peps/pep-0008
 
+Mypy type annotations
+=====================
+
+Certbot uses the ``mypy`` static type checker. Python 3 natively supports official type annotations, which can then be tested for consistency using ``mypy``. Python 3 doesn’t, but type annotations can be `added in comments`_. ``mypy`` does some type checks even without type annotations; we can find bugs in Certbot even without a fully annotated codebase.
+
+Certbot supports both Python 2 and 3, so we’re using Python 2-style annotations.
+
+Zulip wrote a `great guide`_ to using mypy. It’s useful, but you don’t have to read the whole thing to start contributing to Certbot.
+
+To run ``mypy`` on Certbot, use ``tox -e mypy`` on a machine that has Python 3 installed.
+
+Note that instead of just importing ``typing``, due to packaging issues, in Certbot we use the following import statement:
```

Comment:
nit: "...in Certbot we import from acme.magic_typing and have to add some comments for pylint like this:"

My thinking is just to expand and comment on what's actually noteworthy here.